torch/utils/data/datapipes/_typing.py

```diff
@@ -337,6 +337,33 @@ def _hash_(self):
         return hash((self.__name__, self.type))
 
 
+def _simplify_obj_name(obj) -> str:
+    """
+    Simplify the display strings of objects for the purpose of rendering within DataPipe error messages.
+    """
+    if inspect.isfunction(obj):
+        return obj.__name__
+    try:
+        return str(obj.__class__.__qualname__)
+    except Exception as _:
+        return str(obj)
```

Comment:
Wondering why we want to return the class of object rather than object itself?
Like `None` value, the Error message is `NoneType` currently. IMHO, `None` would be more better.

Reply:
It typically gives us a shorter and cleaner name (especially for functions/classes). The `NoneType` behavior may be one of the few cases where it gives a longer name

Reply:
Let me have a look at the behavior of a few more types and see what may be best

Reply:
For example, providing the type for `integer`, `float` and other common data stuctures may causes users more confusion on which `DataPipe` caused such issue.

Reply:
I updated this section of the code to only show the simplified class name when the default string is something like `<very.long.path.to.ClassName object at 0x000000000000>`.

Reply:
Please correct me if I am wrong. Based on the comment in the function, does`<very.long.path.to.ClassName object at 0x000000000000>` only refer to `DataPipe` or refer to any other object?
If it's former, why not we add a `__str__` method to `DataPipe` class to always only return the DataPipe's name?

Reply:
It is being applied to any object. My thinking is that `<...>` isn't very informative and just the class name provide the same amount of information.

We definitely can only apply that logic to `DataPipe` and have that as a `__str__` method.

Reply:
I updated the PR so that simplified object string only affects DataPipes when exceptions are raised within DataPipes.